modules/nf-core/umitools/dedup/main.nf

```diff
@@ -8,11 +8,12 @@ process UMITOOLS_DEDUP {
         'biocontainers/umi_tools:1.1.4--py38hbff2b2d_1' }"
 
     input:
-    tuple val(meta), path(bam), path(bai)
+    tuple val(meta), path(bam, stageAs: 'input/*'), path(bai, stageAs: 'input/*')
```

Comment:
Do these files have to be in a folder for this to work? Otherwise it's preferred to leave them in the top directory

Reply:
There actually is a part about this in the guidelines ([here](https://nf-co.re/docs/contributing/modules#naming-conventions) part 6). In short it tells developers that a check should be in the module which gives a custom error message if the input and output file name are the same. You can easily let the user fix this by letting them adjust the `task.ext.prefix`. This is not something that should be fixed on a module level :)

Reply:
(We've had a lot of discussions about this last year and this is the consensus we reached)

Reply:
great, sounds like i'll be changing the output file to `${prefix}.dedup.bam` and having the output file match `"*.dedup.bam"`. that should work around the initial issue while keeping everything at the top level.

Reply:
No you can keep it like it is now, just add the error message. The pipeline developer can add the `.dedup` suffix to the prefix if they want to themselves, which makes the module way more flexible :)

Reply:
ok, but not the way it is now, because we want stageAs to go away right? if that can stay in then I don't need to add the naming check either...

Reply:
i'm having trouble getting this working. i took out the stageAs options as you suggested, and added the error message. however that still leaves the problem of the output channel matching both bams. however if i change the output channel to:
```
tuple val(meta), path("${prefix}.bam")     , emit: bam
```
it breaks with the following explanation:
```
Caused by:
  Missing output file(s) `null.bam` expected by process `test_umitools_dedup_no_umi:UMITOOLS_DEDUP (test)`
```
because the output declaration can't access `$prefix`. catching issues with name clashing is important but it doesn't solve the original problem i am facing. i don't know if i can follow the naming conventions and keep everything at the top-level while solving my problem. please let me know how i should proceed.

Reply:
You will just need to change the prefix in the `nextflow.config` file in the test directory :) here an example of how another modules does that: https://github.com/nf-core/modules/blob/240937a2a9c30298110753292be041188891f2cb/tests/modules/nf-core/ultra/index/nextflow.config#L7

Also you can just remove the `def` of prefix if you want to use prefix in the output declaration

Most of this information can be found in the modules [guidelines](https://nf-co.re/docs/contributing/modules), it's an interesting read which will answer most of your questions for you :)

Reply:
ok thanks. i tried removing `def` before but got frustrated because it caused other errors related to scope. i seem to have fixed it though and pushed my changes. so now there is no `stageAs` and the issue i noticed before should now be avoided.